pkg/operation/botanist/component/interfaces.go

```diff
@@ -47,12 +47,31 @@ type MigrateWaiter interface {
 	WaitMigrate(ctx context.Context) error
 }
 
+// MonitoringComponent exposes configuration for Prometheus as well as the AlertManager.
+type MonitoringComponent interface {
+	// ScrapeConfig returns the scrape configuration for Prometheus.
+	ScrapeConfig() (string, error)
+	// AlertingRules returns the alerting rules for AlertManager.
```

Comment:
Can you add an explanation for what is expected from the returned map?
```suggestion
	// AlertingRules returns the alerting rules configs for AlertManager (mapping file name to rule config).
```